### PR TITLE
Populate document metadata and autogenerated keys

### PR DESCRIPTION
**Pull request by efirs**

### CODE REVIEW

File: tigris/model.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	"time"
+
+	"github.com/google/uuid"
+	api "github.com/tigrisdata/tigris-client-go/api/server/v1"
+)
+
+// User document model can have its own implementation
+// of this interface to customize metadata handling
+type model interface {
+	SetCreatedAt(t time.Time)
```
Comment (himank) [inline tigris/model.go:28]: model is not exposed but the methods are exposed. Is this intentional?

File: tigris/model.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	// request metadata
+	CreatedAt time.Time `json:"-"`
+	UpdatedAt time.Time `json:"-"`
+	DeletedAt time.Time `json:"-"`
+
+	// This field is unused if user defines its own primary key
+	// If no primary key is defined in the user model
+	// this field is automatically populated with
+	// autogenerated unique UUID
+	ID uuid.UUID `json:"id" tigris:"primary_key,generate"`
```
Comment (himank) [inline tigris/model.go:54]: ah, interesting. Let's coordinate with Jigar as well then this is a cool extension. Also, let's document this.

Comment (himank) [reply tigris/model.go:54]: `autogenerate `

Comment (efirs) [reply tigris/model.go:54]: That's not used at the moment, schema support will be added in the next diff.

File: tigris/model.go (hunk 1 above)
Comment (efirs) [reply tigris/model.go:28]: Will export it as `MetadataSetter`

File: tigris/model.go (hunk 2 above)
Comment (himank) [reply tigris/model.go:54]: that's fine, we can still use the same key name.

**Review by himank — DISMISSED** [review]

**Review by himank — APPROVED** [review]

**Pull request merged**